File: dcp-solutions/2025-01-13/solution.php
```php
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public $val;
 *     public $left;
 *     public $right;
 *     function __construct($x) { $this->val = $x; $this->left = null; $this->right = null; }
 * }
 */
class Solution {
    function findTarget($root, $K) {
        $seen = array();
        return this->findPair($root, $K, $seen);
    }

    private function findPair($node, $sum, &$seen) {
        if (!$node) return null;
        $key = $node->val;
        $other = $sum - $key;

        if (in_array($other, $seen)) {
            return array($other, $key);
        }
        
        $leftResult = $this->findPair($node->left, $sum, $seen);
        if ($leftResult) return $leftResult;
        
        $rightResult = $this->findPair($node->right, $sum, $seen);
        if ($rightResult) return $rightResult;

        array_push($seen, $key);
        return null;
    }
}
```

File: dcp-solutions/2025-01-13/solution.go
```go
// Definition for a binary tree node.
type TreeNode struct {
    Val int
    Left *TreeNode
    Right *TreeNode
}

func findTarget(root *TreeNode, k int) (*TreeNode, *TreeNode) {
    var seen map[int]*TreeNode
    var found func(*TreeNode)
    
    found = func(node *TreeNode) {
        if node == nil {
            return 
        }
        
        if seen != nil && val := seen[node.Val]; *val == node {
            return 
        }
        
        seen[ node.Val ] = node
        
        found( node.Left )
        found( node.Right )
    }

    found(root)

    for _, v := range seen {
        if diff := k - v.Val; other, ok := seen[diff]; ok && *other != v {
            return v, other
        }
    }

    return nil, nil
}
```